kernel/src/common/static_fmt.rs

```diff
@@ -0,0 +1,64 @@
+//! A macro static_fmt! that provides a small amount of static storage
+//! for formatting a string.  The value of the macro has type `&'static str`.
+//! Each invocation overwrites what was written previously.
```

Comment:
Similar to `static_mut_bytes_8`, I think this will only work as expected if the resulting string is used and discarded before any other invocation to `static_fmt!` is called, which is risky.

I wonder if a better solution to error formatting is to just provide an enum with a bunch of cases, where constructors to the enum can take arguments needed for actually formatting a final string, then implement the `Display` trait for the enum.